### PR TITLE
refactor(checkout): CHECKOUT-7449 consolidate all utility tools/files in new packages

### PR DESCRIPTION
**Pull request by richapatel1510**
## What?

1. Created two new packages, `dom-utils` and `error-handling-utils`, to hold common dom/ui utility files and error handling utilities in checkout-js. 

- dom-utils => `getAppliedStyles`, `parseAnchor`, `preventDefault` and `stopPropagation`
- error-handling-utils => `ErrorBoundary` and `ErrorLogger`

2. Update import path across the repo to use all of the above mentioned utilities from related package.

## Why?

In the progress of building checkout monorepos, we copied many utiliy tools to different packages, and it resulted in duplication of those files. So to get rid of those duplicates and to better manage them in the future we moved them to related packages.

## Testing / Proof
CI checks

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/error-handling-utils/project.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "root": "packages/error-handling-utils",
+  "sourceRoot": "packages/error-handling-utils/src",
+  "projectType": "library",
+  "targets": {
+    "lint": {
+      "executor": "@nrwl/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["packages/error-handling-utils/**/*.ts"]
```
Comment (bc-peng) [inline packages/error-handling-utils/project.json:10]:
```suggestion
        "lintFilePatterns": ["packages/error-handling-utils/**/*.{ts,tsx}"]
```

File: packages/dom-utils/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": ["../../.eslintrc.json"],
+  "rules": {
+    "@typescript-eslint/consistent-type-assertions": "off" // 1 error
```
Comment (bc-peng) [inline packages/dom-utils/.eslintrc.json:4]: For one error, does it make sense to use an inline comment to ignore the error?

**Review by bc-peng — APPROVED** [review]
LGTM. 👏

File: packages/error-handling-utils/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": ["../../.eslintrc.json"],
+  "rules": {
+    "prettier/prettier": "off" // 71 errors
```
Comment (animesh1987) [inline packages/error-handling-utils/.eslintrc.json:4]: Can these be auto fixed?

Comment (richapatel1510) [reply packages/error-handling-utils/.eslintrc.json]: yes. Fixed them.

**Review by animesh1987 — APPROVED** [review]

**Comment by bc-peng** [comment]
Sorry, I missed this before.
Could you please update the owner file for the new packages?

**Comment by richapatel1510** [comment]
> Sorry, I missed this before. Could you please update the owner file for the new packages? https://github.com/bigcommerce/checkout-js/blob/master/.github/CODEOWNERS

@bc-peng Done

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -10,6 +10,9 @@
 /packages/payment-integration-api @bigcommerce/team-checkout
 /packages/test-framework @bigcommerce/team-checkout
 /packages/workspace-tools @bigcommerce/team-checkout
+/packages/legacy-hoc @bigcommerce/team-checkout
+/packages/dom-utils @bigcommerce/team-checkout
+/packages/error-handling-utils @bigcommerce/team-checkout
```
Comment (bc-peng) [inline .github/CODEOWNERS:15]: Perhaps we can sort the `* @bigcommerce/team-checkout` list alphabetically.

Comment (richapatel1510) [reply .github/CODEOWNERS]: @bc-peng Sorted.

**Pull request merged**